### PR TITLE
Feature/bitbucket authentication

### PR DESCRIPTION
**Pull request by mminns**
Adds explicit Bitbucket support to GCM, where the host is Bitbucket.org.

Supports Basic Auth and OAuth credentials.
gcm-bbc-login
gcm-bbc-oauth
gcm-bbc-authorize

### CODE REVIEW

File: GitHub.Authentication/CredentialsWindow.xaml.cs [hunk 1]
```diff
@@ -17,5 +17,10 @@ public CredentialsViewModel ViewModel
         {
             get { return DataContext as CredentialsViewModel; }
         }
+
+        private void credentialsControl_Loaded(object sender, System.Windows.RoutedEventArgs e)
```
Comment (Foda) [inline GitHub.Authentication/CredentialsWindow.xaml.cs:21]: Can be removed?

Comment (mminns) [reply GitHub.Authentication/CredentialsWindow.xaml.cs:21]: done

File: Bitbucket.Authentication/OAuth/OAuthAuthenticator.cs [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+                rawUrlData = await SimpleServer.WaitForURLAsync(CallbackUri, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                string message;
+                string details;
+                if (ex.InnerException != null && ex.InnerException.GetType().IsAssignableFrom(typeof(TimeoutException)))
+                {
+                    message = "Timeout awaiting response from Host service.";
+                    details = "Please check the SourceTree's configuration for this Host." + Environment.NewLine +
```
Comment (Foda) [inline Bitbucket.Authentication/OAuth/OAuthAuthenticator.cs:83]: Might want to remove the "SourceTree" part

**Review by Foda — COMMENTED** [review]
Looks good, will pull and test on my machine today.

File: Bitbucket.Authentication/OAuth/OAuthAuthenticator.cs (hunk 2 above)
Comment (mminns) [reply Bitbucket.Authentication/OAuth/OAuthAuthenticator.cs:83]: good point

**Pull request merged**